File: 백준/Bronze/10810. 공 넣기/공 넣기.py

import sys
input = sys.stdin.readline


def solution(n:int, m:int) -> list:
    basket = [0] * (n+1)
    for _ in range(m):
        i, j, k = map(int, input().split())
        for idx in range(i, j+1):
            basket[idx] = k
    return basket[1:]


if __name__ == "__main__":
    n,m = map(int, input().split())
    
    
    print(*solution(n,m))

File: 백준/Bronze/10810. 공 넣기/공 넣기.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);
const arr = Array(N).fill(0);
input.slice(1).forEach(e => {
    const [i, j, k] = e.split(' ').map(Number);
    for (let idx = i-1; idx <j ; idx++){
        arr[idx] = k;
    }
});
console.log(arr.join(' '));